src/ConfigAggregator.php

```diff
@@ -92,6 +96,28 @@ public function getMergedConfig()
         return $this->config;
     }
 
+    /**
+     * Throw an exception if duplicate config providers are found.
+     *
+     * @param ProviderIterable $providers
+     * @return ProviderIterable
+     * @throws InvalidConfigProviderException
+     */
+    private function removeDuplicateProviders(iterable $providers): iterable
```

Comment:
the method is actually validating, not make unique, so method name maybe need change, eg: `verifyDuplicateProviders` or `validateDuplicateProviders`

Reply:
Indeed, with the most recent changes, the name `removeDuplicateProviders` is not the best fit anymore.
I guess `validateDuplicateProviders` should do it.

Pros/cons anybody?